_posts/2018-12-28-Reading-Web-Component-And-Custom-Elements.markdown

```diff
@@ -0,0 +1,26 @@
+---
+layout: post
+title:  "Reading: Make a Native Web Component with Custom Elements v1 and Shadow DOM v1"
+date: "2018-12-28 22:14:00 +0800"
+categories: "reading note"
+author: "Jackson"
+tags: "web component"
+---
+
+[Article Link](https://bendyworks.com/blog/native-web-components)   
+[中文版本](https://zhuanlan.zhihu.com/p/52802048)
+
+#### Summary:
+This article is mainly talking about the web component standards which includes the HTML Templates, Shadow Dom and Custom Elements.
+
+The author made a form component base on the web component's standards, so this component just like native HTML elements, they can have properties, methods, and event listeners. Just the different with React component, web component didn't need any translation engine to translate the code to standard HTML, It just made by vanilla JavaScript and the native HTML. So it doesn't need any dependency. And it will work across modern browsers and it can be used with any JavaScript library or framework that works with HTML. And with the advantage of React component reuse and share easily. And the custom element has lifecycle hooks too, so you can control the component easily.
+
+And with the Shadow Dom's help, we can build the component which defines its internal structure, scoped CSS, and encapsulates your implementation details. It does not interact with the parent DOM. So the parent's style and the script will not affect your component. And the component's style or behavior will not affect the context in which it is used too. And The component will be consistent across all instances of the component, wherever it is used.
```

Comment:
which => that
Generally you use "that" in a sentence that does't use comma (,), and "which" after comma. For example: "I made an app that can play sounds.", and "I made a tough decision last night, which involves avoiding late night snacks."
And The component will be consistent => The component will be consistent... (The "And" is not necessary. It is not a good habit to use too many "and" to connect your sentences).